### DISCUSSION TITLE
Lesson 11: Working with Pinata SDK in Foundry?

### DISCUSSION THREAD

**Discussion by 0xGuvnor** [Cyfrin/foundry-full-course-cu#302] [Q&A] [answered]
In the previous Hardhat version of this course, Patrick wrote some scripts that used the Pinata SDK to programmatically pin an NFT's token URI to IPFS, so I was wondering how would one work with the Pinata SDK, or any other external npm-based SDKs for that matter, with Foundry since JS/TS scripts are not supported (that I know of).

I would like to hear what are the best practices/workarounds people are doing for such cases.

Thanks!

Edit: Based on this [post](https://github.com/Cyfrin/foundry-full-course-f23/discussions/303), it seems that using initialising Foundry and Hardhat within the same project is something that can be done and not an uncommon occurrence. 

So I guess my follow-up question for those who use Foundry and Hardhat in the same project is what are your workflows like? i.e. Do you use Hardhat for running scripts and deploying and Foundry for compiling and testing?

**Comment by Niferu** [comment]
Great question! I would like to hear anything about it too.

**Comment by cromewar** [comment]
Nice question, I would love to know you opinion on this @PatrickAlphaC

**Reply by Niferu** [reply]
Hi @cromewar 

Can you take a look at this please? :D

[link](https://github.com/Cyfrin/foundry-full-course-f23/discussions/295#discussioncomment-6393755)

**Comment by PatrickAlphaC** [comment, marked as answer]
Great question.

So you _could_ write this in solidity as a solidity script, but I don't recommend that at all. This is where a more traditional programming language would come into play. A lot of modern projects are using foundry + hardhat at the moment, and if you like JS, that's a great route to go! The new UniswapV4 for example uses both hardhat & foundry for stuff like this. 

I personally like python for stuff like this.
